README.md

```diff
@@ -1,2 +1,7 @@
 # servant-elm-street
 Generate elm requests, with types, and JSON encoders, and decoders derived by elm-street, given a servant API.
+
+## Encoding headers
```

Comment:
Document the assumption that a a user would never want to use `Maybe` as a header type, because the same effect could be achieved with the `Lenient` modifier.